src/Mod/PartDesign/App/FeatureHelix.cpp

```diff
@@ -71,24 +71,29 @@ const char* Helix::ModeEnums[] = {"pitch-height", "pitch-turns", "height-turns",
 
 PROPERTY_SOURCE(PartDesign::Helix, PartDesign::ProfileBased)
 
+// we purposely use not FLT_MAX because this would not be computable
+const App::PropertyFloatConstraint::Constraints floatTurns = { Base::toDegrees<double>(Precision::Confusion()), INT_MAX, 1.0 };
+const App::PropertyAngle::Constraints floatAngle = { -89.0, 89.0, 1.0 };
```

Comment:
This is a bad restriction as you cannot enter -89.5 or +89.5. I think it's better to set the range [-90.0, +90.0] and then in the execute method raise an error if the angle is too close to the lower or upper limit.

Reply:
I am aware of that. But David restrict it to 89.0 in the dialog too.  I first thought "90 - Precision::Confusion() " might be a better upper limit, but even 89.9 takes ages to compute and I cannot see a real life use case for such a helix beast. Therefore I think david choose 89.0 for good reason and the limits in App and Gui should be equal.

Reply:
@davidosterberg  , what is your opinion?

Reply:
OK but independent of what value range we go for we should not hard-code the limits in the feature **and** the dialog.

Today I have added some convenience functions to more easily get the limits and these functions should then be used for the dialog. The advantage is that whenever we change the range we have to do it only once.

Reply:
I agree that angle > 89deg are not usefu with the current implementation. Would've been nice if 90deg was supported (i.e a flat spiral). Unfortunately makeLongHelix is not general enough.